src/AltinnCore/Common/Services/Implementation/PDFSI.cs

```diff
@@ -62,16 +62,23 @@ public async Task GenerateAndStoreReceiptPDF(Instance instance, UserContext user
             int instanceOwnerId = int.Parse(instance.InstanceOwnerId);
             Guid instanceGuid = Guid.Parse(instance.Id.Split("/")[1]);
             Guid defaultDataElementGuid = Guid.Parse(instance.Data.Find(element => element.ElementType.Equals("default"))?.Id);
+
             Stream dataStream = await _dataService.GetBinaryData(org, app, instanceOwnerId, instanceGuid, defaultDataElementGuid);
             byte[] dataAsBytes = new byte[dataStream.Length];
             dataStream.Read(dataAsBytes);
             string encodedXml = System.Convert.ToBase64String(dataAsBytes);
 
+            byte[] formLayout = _executionService.GetServiceResource(org, app, "FormLayout.json");
+            byte[] textResources = _executionService.GetServiceResource(org, app, "resource.nb-NO.json");
+
+            string formLayoutString = GetUTF8String(formLayout);
+            string textResourcesString = GetUTF8String(textResources);
+
             PDFContext pdfContext = new PDFContext
             {
                 Data = encodedXml,
-                FormLayout = JObject.Parse(_repositoryService.GetJsonFormLayout(org, app)),
-                TextResources = _repositoryService.GetServiceTexts(org, app),
+                FormLayout = JsonConvert.DeserializeObject(formLayoutString),
+                TextResources = JsonConvert.DeserializeObject(textResourcesString),
```

Comment:
formLayoutString and textResourcesString are strings, do you need to DeserializeObject(..) them?

Reply:
Yes. The pdf-generator expects an object, did not work to just send a string.